app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt

```diff
@@ -294,18 +306,75 @@ fun OngoingChallengeCard(
                       .size(48.dp) // Set size to ensure consistency between buttons
               ) {
                 IconButton(
-                    onClick = onDeleteClick,
+                    onClick = { showDialog.value = true },
                     modifier =
                         Modifier.fillMaxSize() // Make the button fill the Box size
                             .testTag("DeleteButton${challenge.challengeId}")) {
                       Icon(
                           imageVector = Icons.Default.Delete,
                           contentDescription = "Delete Challenge",
-                          tint = Color.Gray, // Explicitly set color for visibility
+                          tint = MaterialTheme.colorScheme.error,
                           modifier = Modifier.size(30.dp) // Icon size for better visibility
                           )
                     }
               }
         }
   }
+
+  // Confirmation dialog
+  ConfirmationDialog(showDialog = showDialog, onClickAction = onDeleteClick)
+}
+
+@SuppressLint("SuspiciousIndentation")
+@Composable
+fun ConfirmationDialog(showDialog: MutableState<Boolean>, onClickAction: () -> Unit) {
```

Comment:
## Requested changes

This is too much to duplicate around, you should move this and the delete button into `Buttons.kt`, and create a new `@Composable` called `DialogButton()` or `DeleteButton()` or something.